docs/source/plugin/what.rst

```diff
@@ -0,0 +1,135 @@
+=================
+What is a Plugin?
+=================
+
+Before writing your first plugin, you may want to know what is a plugin, how is
+it made, and what are the best solutions for your need. In this chapter we'll
+cover what is a plugin, what you can do with them, and some vocabulary.
+
+You may want to skip ahead to :doc:`anatomy` if you already know what a plugin
+is and what types of plugin there are.
+
+.. contents::
+    :local:
+    :depth: 2
+
+Plugin Types
+============
+
+There are four types of plugins: :term:`Single file plugin`,
+:term:`Folder plugin`, :term:`Namespace package plugin`, and
+:term:`Entry point plugin`.
+
+Single file
+-----------
+
+A **Single file** plugin is the most basic form a Sopel plugin can take. It is
+composed of a single file available to Sopel from its plugins directory (or
+from one of its
+:attr:`extra directories <sopel.config.core_section.CoreSection.extra>`). You
+should write a single file plugin when:
+
+* everything is in a single file
+* you don't have any dependencies other than Sopel and the Python's built-in
+  library
+
+This type of plugin is easy to install and share: copy its file into the right
+location, and Sopel will load it on startup.
```

Comment:
```suggestion
This type of plugin is easy to install: copy its file into the right location,
and Sopel will load it on startup.
```

Goes with my "alternative suggestion" for the above list of reasons to use single-file plugins.